### PR TITLE
Create Facility on Facility Form

### PR DESCRIPTION
**Pull request by meganfung38**
# Checklist

- [x] Have I clearly explained the issue or feature request that this PR addresses?
- [x] Have I provided where new implementations for the PR have been made?
- [x] Did I highlight any areas that may need special attention during the review?
- [x] If applicable, did I specify any setup requirements or dependencies for testing?
- [x] Have I disclosed any known issues or limitations related to this PR? Did I explain why these limitations exist and how they might be addressed in the future?
- [x] Have I provided enough context for someone unfamiliar with the issue or the codebase to understand the changes?
- [x] Have I assigned appropriate reviewers who are knowledgeable about the changes? Did I consider including team members who might be affected or interested in the changes?

# Description

## Summary
Issue #145:
- original implementation did not pass facility form inputs to backend
- new implementation creates a facility in the db using the facility form inputs
- facilities can now be made using the form and will render as a marker on the main map

## Expectations
- Steps for Testing: 
   1. Navigate to the facility form
   2. Wait for the map view to render-- implementation will not allow for the creation of a facility if current location has not yet been fetched (markerAddress must have a value)
   3. Fill in the form
   4. Submit the facility
       - Expect to navigate back to the main screen 
       - The main map should now reload with the newly created facility 
       - if you click the facility marker, expect to navigate to its facility profile where you should see your inputs:
           - Name of Facility: address of marker 
           - Image Carousel: should populate with your uploaded images 
           - Add Review button: should navigate to review form but implementation to render facility name has not been done yet
           - Star ratings should by default be at 5 stars
           - Operation Hours, Category (should be "None"), Tags chosen, and description input should display in collapse view

## Additional Comments/Notes
** May need to reload main to see new facility-- sometimes works sometimes does not 
- Controller Function, createBathroom is now in bathroomController.tsx-- was originally in userController.tsx
- Made some edits to createBathroom to correctly pass in the required parameters for the creation of a facility

### CODE REVIEW

File: controllers/userController.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import axios from "axios";
 import { User as UserProfile } from "../types/user";
-import { Facility as BathroomProfile } from "../types/facility";
+// import { Facility as BathroomProfile } from "../types/facility";
```
Comment (lindlands) [inline controllers/userController.ts:3]: Why is this commented out?

File: screens/facility-profile.tsx [hunk 2]
```diff
@@ -69,6 +73,83 @@ type FacilityProfileRouteProp = RouteProp<
   "FacilityProfile"
 >;
 
+const imageUrls = [
```
Comment (lindlands) [inline screens/facility-profile.tsx:76]: Is this PR waiting on the image carousel PR to get merged?

File: controllers/userController.ts [hunk 3]
```diff
@@ -55,56 +55,58 @@ export async function getUserProfile(UserID: string): Promise<UserProfile> {
...
+//  * @param {string} Operations - Bathroom's hours of operation
+//  * @param {string[]} Reviews - Reviews on bathroom
+//  * @param {Date} Date - Date the bathroom was created/posted
+//  * @param {string[]} PictureURL - Picture(s) of the bathroom
+//  * @param {number} RatingAVG - The rating average of the bathroom
+//  * @param {number} Favorites - The number of favorites the bathroom has
+//  * @param {number} Reports - The number of reports on the bathroom
+//  * @returns {Promise<string>} - Returns a promise with a success message upon successfully creating a new bathroom
+//  */
+// export async function createBathroom(
```
Comment (lindlands) [inline controllers/userController.ts:72]: Why is this also commented out?

File: controllers/userController.ts (hunk 1 above)
Comment (meganfung38) [reply controllers/userController.ts:3]: oops let me remove all comments-- thanks for catching these

File: screens/facility-profile.tsx (hunk 2 above)
Comment (meganfung38) [reply screens/facility-profile.tsx:76]: this PR already includes the image carousel PR-- will delete this

**Comment by dylantompkins** [comment]
Could the following be relevant to why updating the data on main after submitting a new facility is inconsistent?

https://reactnavigation.org/docs/use-focus-effect/#running-asynchronous-effects

**Comment by meganfung38** [comment]
@dylantompkins Thanks for the article-- my original commit already used the useFocusEffect, but main was still only updating sometimes. Read through the article and made some minor changes in the latest commit-- hopefully it is fixed, when I tested just now it main map was updating.
Changes are on lines 26-30 of main.tsx

**Review by JonasSchiessl — CHANGES_REQUESTED** [review]
can you fix [this](https://sonarcloud.io/project/issues?open=AY_khTmzTVfQNxUMSCCq&id=Green-Kittens_bathroom-finder-frontendl) code smell. The time selector does not seem to be working.

**Comment by meganfung38** [comment]
@JonasSchiessl the link to the code smell doesn't work-- the time selector works on my end, what problem are you encountering?

**Comment by JonasSchiessl** [comment]
> @JonasSchiessl the link to the code smell doesn't work-- the time selector works on my end, what problem are you encountering?

The one that's not the controller one specifically using .? Instead of &&

**Comment by JonasSchiessl** [comment]
> @JonasSchiessl the link to the code smell doesn't work-- the time selector works on my end, what problem are you encountering?

[Prefer using an optional chain expression instead, as it's more concise and easier to read.](https://sonarcloud.io/project/issues?pullRequest=165&open=AY_khTmzTVfQNxUMSCCq&id=Green-Kittens_bathroom-finder-frontend)

**Review by JonasSchiessl — APPROVED** [review]
lgtm, great job.

**Pull request merged**